stock_release_channel_shipment_advice/models/shipment_advice_planner.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ShipmentAdvicePlanner(models.TransientModel):
+
+    _inherit = "shipment.advice.planner"
+
+    release_channel_id = fields.Many2one(
+        comodel_name="stock.release.channel", string="Release Channel"
+    )
+    picking_to_plan_ids = fields.Many2many(
+        domain='[("can_be_planned_in_shipment_advice", "=", True),'
+        '("release_channel_id", "=?", release_channel_id),]',
+        compute="_compute_picking_to_plan_ids",
+        store=True,
+        readonly=False,
+    )
+
+    @api.depends("release_channel_id")
+    def _compute_picking_to_plan_ids(self):
+        for rec in self:
+            if rec.release_channel_id:
+                rec.picking_to_plan_ids = rec.release_channel_id.picking_to_plan_ids
```

Comment:
@sbejaoui I would remove the compute. The `picking_to_plan_ids`will be passed to the planner by the release channel on call

Reply:
The compute is used as an onchange for the release channel

Reply:
The problem here is that we diverge from the initial implementation.... If a new addon add a new criteria to compute this list it should inherit from this addon since the compute method is not implemented into the base addon.

If we want to keep the compute method, it should be declared into the base addon with readonly=False and do nothing and extended here. Moreover, the depends should call a method from the model to get the list of fields it depends on to ease the extension....

in shipping_advice_planner
```python

@api.model
def _get_compute_picking_to_plan_ids_depends(self):
    return []

@api.depends(lambda m: m._get_compute_picking_to_plan_ids_depends())
def _compute_picking_to_plan_ids(self):
    pass

```

here
```python
@api.model
def _get_compute_picking_to_plan_ids_depends(self):
    depends = super()._get_compute_picking_to_plan_ids_depends()
    depends += ["release_channel_id"]

@api.depends(lambda m: m._get_compute_picking_to_plan_ids_depends())
def _compute_picking_to_plan_ids(self):
    res = super()_compute_picking_to_plan_ids()
    for rec in self:
        if rec.release_channel_id:
            rec.picking_to_plan_ids = rec.release_channel_id.picking_to_plan_ids

```